source/_posts/2017/2017-09-08-how-to-deprecate-code-functionality -in-a-symfony-way.md

```diff
@@ -212,92 +212,8 @@ I said...
 
 
 
-## Bonus: Gradual Deprecation
+That was [Symfony's Backward Compatibility Promise](https://symfony.com/doc/current/contributing/code/bc.html) in a nutshell.
 
-You can see [the same technique in Symfony](https://github.com/symfony/symfony/blob/c9bee848c5e009de190a8cc4ca85948a93ded129/src/Symfony/Component/Config/Definition/ArrayNode.php#L254) with 1 difference: 
-
-```php
-@trigger_error('Method add() is deprecated, use addHtml() instead.', E_USER_DEPRECATED);
-```
-
-Notice `@`!
-
-So the error is there for the user, but it is suppressed? WTF!
-
-
-### Step by Step
-
-It's related to release process. Just imagine this life cycle of `add()` method:
- 
-- 3.0 - method `add()` was added
-- 3.1 - method `add()` will be replace by `addHtml()`, but it's BC break, so `add()` will be here until 4.0
-- 3.2 - **both methods should work with no error** - `@` sign helps with that
-- 3.3 - **`add()` won't work from next version** (major), so we remove `@` to inform the user
-- 4.0 - only `addHtml()`
-   
-Now in code:
-
-
-**3.0**
-
-```php
-public function add(...)
-{
-}
-```
-
-**3.1**
-
-```php
-public function add(...)
-{
-    @trigger_error('Method add() is deprecated, use addHtml() instead.', E_USER_DEPRECATED);
-    $this->addHtml(...);
-}
-
-public function addHtml(...)
-{
-}
-```
-
-**3.2** (same as 3.1)
-
-```php
-public function add(...)
-{
-    @trigger_error('Method add() is deprecated, use addHtml() instead.', E_USER_DEPRECATED);
-    $this->addHtml(...);
-}
-
-public function addHtml(...)
-{
-}
-```
-
-**3.3**
-
-```php
-public function add(...)
-{
-    trigger_error('Method add() is deprecated, use addHtml() instead.', E_USER_DEPRECATED);
-    $this->addHtml(...);
-}
-
-public function addHtml(...)
-{
-}
-```
-
-**4.0**
-
-```php
-public function addHtml(...)
-{
-}
-```
```

Comment:
I see the cyclomatic complexity of this post would be like 10x higher with this section.

Thanks @enumag for prescious feedback! Feels much better to read now